ta_services/secure_key_services/include/sks_ta.h

```diff
@@ -14,13 +14,615 @@
+ * object attributes serialized one after the other. The header defines the
+ * number of attributes of the object. Each attribute is defined by 3 serialized
+ * fields, see struct sks_reference.
+ */
+
+/*
+ * sks_object_head - Header of object whose data are serialized in memory
+ *
+ * @blobs_size - byte size of the serialized data
+ * @blobs_count - number of items in the blob
+ * @blobs - then starts the blob binary data
+ */
+struct sks_object_head {
+	uint32_t blobs_size;
+	uint32_t blobs_count;
+	uint8_t blobs[];
+};
+
+/*
+ * Attribute reference in the TA ABI
+ *
+ * @id - the 32bit identificator of the attribute, see SKS attribute IDs
+ * @size - the 32bit value attribute byte size
+ * @data - location defines base memory of the attribute effective value
+ */
+struct sks_reference {
+	uint32_t id;
+	uint32_t size;
+	uint8_t data[];
+};
+
+/*
+ * SKS_CMD_DESTROY_OBJECT - Destroy an object
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ *			   [uint32_t object_handle]
+ * param#1: none
+ * param#2: none
+ * param#3: none
+ *
+ * This commands relates to the PKCS#11 API function C_DestroyObject().
+ */
+#define SKS_CMD_DESTROY_OBJECT		0x0000000f
+
+/*
+ * SKS_CMD_ENCRYPT_INIT - Initialize decryption processing
+ * SKS_CMD_DECRYPT_INIT - Initialize encryption processing
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ *			   [struct sks_reference proc + proc parameters data]
+ * param#1: none
+ * param#2: none
+ * param#3: none
+ *
+ * These commands relate to the PKCS#11 API functions C_EncryptInit() and
+ * C_DecryptInit.
+ */
+#define SKS_CMD_ENCRYPT_INIT		0x00000010
+#define SKS_CMD_DECRYPT_INIT		0x00000011
+
+/*
+ * SKS_CMD_ENCRYPT_UPDATE - Update encryption processing
+ * SKS_CMD_DECRYPT_UPDATE - Update decryption processing
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ * param#1: in-memref : [input-data]
+ * param#2: out-memref : [output-data]
+ * param#3: none
+ *
+ * These commands relate to the PKCS#11 API functions C_EncryptUpdate() and
+ * C_DecryptUpdate.
+ */
+#define SKS_CMD_ENCRYPT_UPDATE		0x00000012
+#define SKS_CMD_DECRYPT_UPDATE		0x00000013
+
+/*
+ * SKS_CMD_ENCRYPT_FINAL - Finalize encryption processing
+ * SKS_CMD_DECRYPT_FINAL - Finalize decryption processing
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ * param#1: none
+ * param#2: out-memref : [output-data]
+ * param#3: none
+ *
+ * These commands relate to the PKCS#11 API functions C_EncryptFinal() and
+ * C_DecryptFinal.
+ */
+#define SKS_CMD_ENCRYPT_FINAL		0x00000014
+#define SKS_CMD_DECRYPT_FINAL		0x00000015
+
+/*
+ * SKS_CMD_GENERATE_SYMM_KEY - Generate a symmetric key
+ *
+ * param#0: in(*)-memref : [uint32_t session_handle]
+ *			   [struct sks_reference proc + proc parameters data]
+ *			   [struct sks_object_head attribs + attributes data]
+ * param#1: none
+ * param#2: out-memref : [uint32_t object_handle]
+ * param#3: none
+ *
+ * This command relates to the PKCS#11 API functions C_GenerateKey() and
```

Comment:
s/This command relates to the PKCS#11 API functions C_GenerateKey() and C_DecryptInit./This command relates to the PKCS#11 API functions C_GenerateKey().

Reply:
thanks